src/features/lightspeed/contentMatchesWebview.ts

```diff
@@ -107,22 +108,82 @@ export class ContentMatchesWebview implements vscode.WebviewViewProvider {
     this._view.webview.html = "";
   }
 
+  public isError(
+    contentMatchResponses: ContentMatchesResponseParams | IError
+  ): contentMatchResponses is IError {
+    return (contentMatchResponses as IError).code !== undefined;
+  }
+
   private async getWebviewContent(): Promise<string> {
-    const noContentMatchesFoundHtml = `<html><body>No training matches found for the latest accepted suggestion.</body></html>`;
+    const noActiveSuggestionHtml = `
+      <html>
+        <body>
+          <p>Training matches cannot be retrieved. No active suggestion found.</p>
+        </body>
+      </html>`;
     if (
       this.suggestionDetails.length === 0 ||
       !this.suggestionDetails[0].suggestion
     ) {
-      return noContentMatchesFoundHtml;
+      return noActiveSuggestionHtml;
     }
+
     const suggestion = this.suggestionDetails[0].suggestion;
     const suggestionId = this.suggestionDetails[0].suggestionId;
-
     const contentMatchResponses = await this.requestInlineSuggestContentMatches(
       suggestion,
       suggestionId
     );
     this.log(contentMatchResponses);
+
+    if (this.isError(contentMatchResponses)) {
+      return this.getErrorWebviewContent(contentMatchResponses);
+    } else {
+      return this.getContentMatchWebviewContent(
+        suggestion,
+        contentMatchResponses
+      );
+    }
+  }
+
+  private async getErrorWebviewContent(error: IError) {
+    let detail: unknown = error.detail;
+    if (typeof error.detail === "string") {
+      detail = error.detail as string;
+    } else if (typeof error.detail === "object") {
+      detail = JSON.stringify(error.detail, undefined, "  ");
+    }
+
+    const errorHtml = `
+      <html>
+        <head>
+          <!-- https://code.visualstudio.com/api/extension-guides/webview#content-security-policy -->
+          <meta http-equiv="Content-Security-Policy" content="default-src 'none';">
+        </head>
+        <body>
+        <p>An error occurred trying to retrieve the training matches.</p>
+        <p><b>Message:</b> ${error.message}</p>
+        <details>
+          <summary><b>Detail:</b></summary>
+          <p>${detail}</p>
```

Comment:
We should exclude the details section if we don't have any. A 503 from contentmatches currently gives me this:
![image](https://github.com/ansible/vscode-ansible/assets/3698180/f724b510-ab3c-420d-a3e1-5346bafa2d70)

Reply:
Shall we skip the lower right pop-up when we're already displaying the error in the panel?

Reply:
Fair point. Removed.